src/runtime/virtcontainers/clh.go

```diff
@@ -1294,6 +1295,19 @@ func (clh *cloudHypervisor) bootVM(ctx context.Context) error {
 		return fmt.Errorf("VM state is not 'Created' after 'CreateVM'")
 	}
 
+	// This situation would never actually happen, as the network devices
+	// are created before actually creating the VM.
+	// We need, however, to add the check here in order to make our tests
+	// happy enough.
+	if clh.netDevices != nil {
```

Comment:
Not sure why we need this if it is only for tests. Can you clarify what test fails without this. If this is just test-specific code, maybe we need to modify our test itself.

Reply:
@amshinde, this is done as it's so we can avoid mocking the whole bootVM function in the very first commits.
Later in the series, this is removed as we end up mocking the vmAddNetPutRequest() function.

Does this sound reasonable?